### DISCUSSION TITLE
separating critical js to reduce renderblocking resources and cause site to load faster initially.

### DISCUSSION THREAD

**Discussion by SQLJames** [razonyang/hugo-theme-bootstrap#234] [Ideas]
I am not sure how this can be done but it looks like we have java script that is used for mobile and some used for desktop that is always loaded. Is there a way to identify what is the critical JavaScript and load that first vs a deferred load of JavaScript on the noncritical items. I noticed that my site is a lot slower when loading on mobile. It turns out script evaluation according to a lighthouse test is about 2 seconds of that loading time. 
This is eval with the main site (https://blog.rhoat-homelab.com/)
![image](https://user-images.githubusercontent.com/22476733/137647945-ba90035b-8867-4248-8dfe-1b33bb35e5dd.png)
vs the eval on a post I did. (https://blog.rhoat-homelab.com/posts/golang/interfaces/)
![image](https://user-images.githubusercontent.com/22476733/137647929-09236930-835b-4e18-89dd-427db96a2855.png)
![image](https://user-images.githubusercontent.com/22476733/137647932-1aceb173-e0b5-4d2a-ad11-9cf90ee518b1.png)

Do we have any ideas on how I can improve this?

**Comment by razonyang** [comment]
Currently, all JS components(except katex.js, mermaid.js, search.js) were bundled into one for reducing HTTP requests.

There are three key components, Bootstrap built-in components, dark/light mode component and the palette component. The rest of the components are not critical. So we can divide those components into two, but the downside is that, it increases HTTP requests. I don't know if this method is effective.

**Comment by zwoop** [comment]
Trying to figure out how to disable katex and mermaid permanently, I don't need either feature, and would like to save those downloads. I can't find any configs to turn these off ?

**Reply by razonyang** [reply]
There are two parameters called math and diagram, both of them are turn off  by default.

**Reply by zwoop** [reply]
Found it, thanks! :)

**Comment by razonyang** [comment]
I think the issue has been fixed.

https://pagespeed.web.dev/report?url=https%3A%2F%2Fhbs.razonyang.com%2Fen%2F

**Comment by zwoop** [comment]
Yeh, this removed all the layout shifts etc. for me as well, my site now shows 100/100 on the pagespeed tests as well!

**Comment by SQLJames** [comment]
Sorry I have been MIA on this thread, I see that the latest release also fixed my site, however, the mobile side still has some of these resources being blocked. Did you end up going with your original thought on this fix? It looks like there was alot of changes in your merge.
